Fixes part of #6240: Fix Flaky Exploration Card Finding

U8NWXD opened the pull request:
## Explanation
Finding exploration cards on the creator dashboard page was flaky
because sometimes the protractor test would search for the cards before
the cards had loaded. Further, the logic used to identify matching cards
failed when the card contained text besides the exploration title. This
commit resolves these issues by:

* Waiting for an exploration card to appear before searching for them
* When filtering cards by title, consider cards to match if their text
  equals the target title or begins with the target title concatenated
  with a newline. This works because when other text is included in the
  card, a newline separates that text from the title.

Fixes part of #6240 

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python -m scripts.pre_commit_linter` and `python -m scripts.run_frontend_tests`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR has an appropriate "PR CHANGELOG: ..." label (If you are unsure of which label to add, ask the reviewers for guidance).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [x] The PR addresses the points mentioned in the codeowner checks for the files/folders changed. (See the [codeowner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).)
- [x] The PR is **assigned** to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer and don't tick this checkbox.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue. Do not only request the review but also add the reviewer as an assignee.

U8NWXD committed:
Protractor: Fix Flaky Exploration Card Finding

Finding exploration cards on the creator dashboard page was flaky
because sometimes the protractor test would search for the cards before
the cards had loaded. Further, the logic used to identify matching cards
failed when the card contained text besides the exploration title. This
commit resolves these issues by:

* Waiting for an exploration card to appear before searching for them
* When filtering cards by title, consider cards to match if their text
  equals the target title or begins with the target title concatenated
  with a newline. This works because when other text is included in the
  card, a newline separates that text from the title.

U8NWXD committed:
Style fixes

nithusha21 commented on core/tests/protractor_utils/waitFor.js:
```diff
@@ -84,9 +84,20 @@ var visibilityOf = function(element, errorMessage) {
+ *                                 retrieves an element, e.g.
+ *                                 element(by.css(...))
+ */
+var elementToExist = function(elementFinder) {
```
How is this different from visibilityOf?

U8NWXD replied:
This just checks that the element exists in the DOM, not that it's visible. In this case, I agree that visibility is more what we're after, so I'll remove this and replace its usage with visibilityOf

U8NWXD replied:
Done in e630f5ae2

nithusha21 reviewed (commented):
LGTM overall. Just one question!

U8NWXD committed:
Replace presenceOf with visibilityOf

We were waiting for a card to exist in the DOM (whether or not it was
visible), but we really do want it to be visible before clicking it.
Further, by relying on `visibilityOf`, we can use existing code. This
commit makes this switch.

U8NWXD commented:
@nithusha21 PTAL

seanlip commented:
@kevinlee12 @nithusha21 PTAL as codeowners. Thanks!

kevinlee12 reviewed (approved):
lgtm!
